model/generate_test.go

```diff
@@ -1182,6 +1251,48 @@ func (s *GenerateSuite) TestSaveNewTasksWithDependencies() {
 	s.Equal(4, generatorTask.NumGeneratedTasks)
 }
 
+func (s *GenerateSuite) TestSaveNewTasksWithDependenciesInNewBuilds() {
+	generator := task.Task{
+		Id:          "task_that_called_generate_task",
+		DisplayName: "task_that_called_generate_task",
+		Version:     "version_that_called_generate_task",
+		BuildId:     "sample_build",
+		Activated:   true,
+	}
+	s.NoError(generator.Insert())
+	sampleBuild := build.Build{
+		Id:           "sample_build",
+		BuildVariant: "a_variant",
+		Version:      "version_that_called_generate_task",
+	}
+	v := &Version{
+		Id:            "version_that_called_generate_task",
+		BuildIds:      []string{"sample_build"},
+		BuildVariants: []VersionBuildStatus{},
+	}
+	pp := &ParserProject{}
+	err := util.UnmarshalYAMLWithFallback([]byte(projYmlTwoBVs), &pp)
+	s.NoError(err)
+	pp.Id = "version_that_called_generate_task"
+	s.NoError(pp.Insert())
+	s.NoError(sampleBuild.Insert())
+	s.NoError(v.Insert())
+
+	g := generatedProjectWithNewBuild
+	g.Task = &generator
+	p, pp, err := FindAndTranslateProjectForVersion(s.ctx, s.env.Settings(), v, false)
+	s.Require().NoError(err)
+	p, pp, v, err = g.NewVersion(context.Background(), p, pp, v)
```

Comment:
Use `s.ctx` instead of making a `context.Background()` (also it's generally best not to use a plain `context.Background()` as-is except in very special cases).